wis2box-management/wis2box/metadata/discovery.py

```diff
@@ -176,6 +177,10 @@ def publish(ctx, filepath, verbosity):
     try:
         dm = DiscoveryMetadata()
         record_mcf = dm.parse_record(filepath.read())
+
+        if record_mcf['wis2box']['topic_hierarchy'] not in DATADIR_DATA_MAPPINGS['data'].keys():  # noqa
+            raise click.ClickException('No matching topic hierarchy')
```

Comment:
exception = f"topic_hierarchy={meta['topic_hierarchy']} not found in data-mappings\n"
exception += f"topic_hierarchy defined in data-mappings: {DATADIR_DATA_MAPPINGS['data'].keys()}" 
raise click.ClickException(exception)